pystac/item.py

```diff
@@ -476,14 +476,9 @@ def from_dict(
             assets={k: Asset.from_dict(v) for k, v in assets.items()},
         )
 
-        has_self_link = False
         for link in links:
-            has_self_link |= link["rel"] == pystac.RelType.SELF
             item.add_link(Link.from_dict(link))
 
-        if not has_self_link and href is not None:
-            item.add_link(Link.self_href(href))
-
```

Comment:
I don't disagree with your changes here in the sense that it remove duplicate logic contained in the constructor (use the `href` to create a `self` link). But this now deleted logic did not add a duplicate `self` link. If you test this revised code with the same example given in the issue, you still get two `self` links. 

I think the problem exists when you create an Item from an href where the Item contains a `self` link. The href is being passed to the constructor, which adds a new `self` link from that href. After using the constructor (which added a new `self` link), the `from_dict` method then adds all the existing links, which also includes the `self` link. So you end up with two `self` links. There needs to be a check for an existing self link in this for loop. But not sure what to do if you end up with two different values.

Reply:
Good catch, thanks. I elected to prefer the `href`-generated `self` link, rather than the one that came from the original item.